File: Find the fine/Find_the_fine.py
class Solution:
    def totalFine(self, date, car, fine):
        ans = 0
        is_even_date = (date % 2 == 0)
        
        for i in range(len(car)):
            if is_even_date and car[i] % 2 == 1:
                ans += fine[i]
            elif not is_even_date and car[i] % 2 == 0:
                ans += fine[i]
        
        return ans

    
  

File: Find the fine/Find_the_fine.java
class Solution {

    public long totalFine(int date, int car[], int fine[]) {
        long ans = 0;
        boolean isEvenDate = (date % 2 == 0);
        int n = car.length;

        for (int i = 0; i < n; i++) {
            if (isEvenDate && car[i] % 2 == 1) {
                ans += fine[i];
            } else if (!isEvenDate && car[i] % 2 == 0) {
                ans += fine[i];
            }
        }
        return ans;
    }
}